packages/editor/src/components/entities-saved-states/hooks/use-is-dirty.js

```diff
@@ -4,48 +4,44 @@
 import { useSelect } from '@wordpress/data';
 import { store as coreStore } from '@wordpress/core-data';
 import { useMemo, useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
-
-const TRANSLATED_SITE_PROPERTIES = {
-	title: __( 'Title' ),
-	description: __( 'Tagline' ),
-	site_logo: __( 'Logo' ),
-	site_icon: __( 'Icon' ),
-	show_on_front: __( 'Show on front' ),
-	page_on_front: __( 'Page on front' ),
-	posts_per_page: __( 'Maximum posts per page' ),
-	default_comment_status: __( 'Allow comments on new posts' ),
-};
 
 export const useIsDirty = () => {
-	const { editedEntities, siteEdits } = useSelect( ( select ) => {
-		const { __experimentalGetDirtyEntityRecords, getEntityRecordEdits } =
-			select( coreStore );
+	const { editedEntities, siteEdits, siteEntityConfig } = useSelect(
+		( select ) => {
+			const {
+				__experimentalGetDirtyEntityRecords,
+				getEntityRecordEdits,
+				getEntityConfig,
+			} = select( coreStore );
 
-		return {
-			editedEntities: __experimentalGetDirtyEntityRecords(),
-			siteEdits: getEntityRecordEdits( 'root', 'site' ),
-		};
-	}, [] );
+			return {
+				editedEntities: __experimentalGetDirtyEntityRecords(),
+				siteEdits: getEntityRecordEdits( 'root', 'site' ),
+				siteEntityConfig: getEntityConfig( 'root', 'site' ),
+			};
+		},
+		[]
+	);
 
 	const dirtyEntityRecords = useMemo( () => {
 		// Remove site object and decouple into its edited pieces.
 		const editedEntitiesWithoutSite = editedEntities.filter(
 			( record ) => ! ( record.kind === 'root' && record.name === 'site' )
 		);
 
+		const siteEntityLabels = siteEntityConfig?.meta?.labels ?? {};
 		const editedSiteEntities = [];
 		for ( const property in siteEdits ) {
 			editedSiteEntities.push( {
 				kind: 'root',
 				name: 'site',
-				title: TRANSLATED_SITE_PROPERTIES[ property ] || property,
+				title: siteEntityLabels[ property ] || property,
 				property,
 			} );
 		}
 
 		return [ ...editedEntitiesWithoutSite, ...editedSiteEntities ];
-	}, [ editedEntities, siteEdits ] );
+	}, [ editedEntities, siteEdits, siteEntityConfig ] );
```

Comment:
Unrelated to this PR, it feels like we could do the same as we do for "site" for all entities. (Show edited properties separately using the labels)